extension/src/pipeline/data.ts

```diff
@@ -1,27 +1,50 @@
+import { dirname } from 'path'
 import { AvailableCommands, InternalCommands } from '../commands/internal'
 import { BaseData } from '../data'
+import { findFiles } from '../fileSystem/workspace'
 
-export class PipelineData extends BaseData<string> {
+export class PipelineData extends BaseData<{
+  dag: string
+  stages: { [pipeline: string]: string | undefined }
+}> {
   constructor(dvcRoot: string, internalCommands: InternalCommands) {
     super(
       dvcRoot,
       internalCommands,
       [{ name: 'update', process: () => this.update() }],
       ['dvc.yaml']
     )
-
-    void this.managedUpdate()
   }
 
   public managedUpdate() {
     return this.processManager.run('update')
   }
 
   public async update(): Promise<void> {
-    const dag = await this.internalCommands.executeCommand(
-      AvailableCommands.DAG,
-      this.dvcRoot
-    )
-    return this.notifyChanged(dag)
+    const [dag, fileList] = await Promise.all([
+      this.internalCommands.executeCommand(AvailableCommands.DAG, this.dvcRoot),
+      this.findDvcYamls()
```

Comment:
[F] Unfortunately `dvc stage list -R .` does not fail for broken `dvc.yaml`s (that fail validation). We need that information because we've updated the logic to count invalid `dvc.yaml`(s) towards there being stages/pipelines. This is because we don't want to show the `Add Stage` button under the circumstances that there is a pipeline that has failed validation. There should already be enough information in the experiments table to work that out.

Reply:
so the logic is now 

1. find all `dvc.yaml`s for the project.
2. run `dvc stage list` against each directory
3. return an object of the format `{ [dir:string]: stdout }` to the model for processing